### DISCUSSION TITLE
How to create a custom combox with search filter option

### DISCUSSION THREAD

**Discussion by roshannzth** [microsoft/microsoft-ui-xaml#9053] [Q&A]
I am trying to create a custom combox like shown in the below image
**Closed state**
<img src="https://github.com/microsoft/microsoft-ui-xaml/assets/65062002/fca75292-1bdf-4b55-8ccc-b928ddba3fc3"  />

**Opened state**
<img src="https://github.com/microsoft/microsoft-ui-xaml/assets/65062002/193f7915-2e68-468d-a25a-f1ddf1d0ac84" height="400" />

These are the needed workflow for the opened state
1)ComboBox popup comes below the comboBox control.
2)1st item in the comboBox is a search filter when adding text into it,
 the popup list can increase or decrease with matched results.
 Note: You can't select the search box item, 
 
 what approach is advised here? is this even possible?

**Comment by ghost1372** [comment]
i think you can use AutoSuggestBox

**Reply by roshannzth** [reply]
I probably need to copy the style of Combo Box defined in generic.xaml
and customize it by adding an autosuggestion box and tweak the behavior of the popup as per my needs.

**Reply by lostluna1** [reply]
> 我认为您可以使用 AutoSuggestBox

I'm glad to see your reply here. I used to have the same question as the questioner, but now I've noticed AutoSuggestBox. Maybe I should use it vigorously

**Reply by roshannzth** [reply]
> > 我认为您可以使用 AutoSuggestBox
> 
> I'm glad to see your reply here. I used to have the same question as the questioner, but now I've noticed AutoSuggestBox. Maybe I should use it vigorously

Yes, AutoSuggestBox by default, or create a custom textbox and override needed functionalities
